core/src/apps/cardano/sign_tx.py

```diff
@@ -586,51 +723,112 @@ async def _show_output(
     protocol_magic: int,
     network_id: int,
 ) -> None:
-    if output.address_parameters:
-        await _fail_or_warn_if_invalid_path(
-            ctx,
-            SCHEMA_ADDRESS,
-            output.address_parameters.address_n,
-            CHANGE_OUTPUT_PATH_NAME,
-        )
+    if signing_mode == CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER:
+        return
 
-        await _show_change_output_staking_warnings(
-            ctx, keychain, output.address_parameters, output.amount
-        )
+    if output.asset_groups_count > 0:
+        await show_warning_tx_output_contains_tokens(ctx)
+
+    is_change_output = False
+    if address_parameters := output.address_parameters:
+        is_change_output = True
+
+        address_policy = get_address_policy(keychain, address_parameters)
+        if address_policy == ADDRESS_POLICY_SHOW_SPLIT:
+            payment_credential_policy = get_output_payment_credential_policy(
+                address_parameters, signing_mode
+            )
+            stake_credential_policy = get_output_stake_credential_policy(
+                keychain, address_parameters, signing_mode
+            )
 
-        if _should_hide_output(output.address_parameters.address_n):
-            return
+            await show_set_output_payment_credential(
+                ctx,
+                address_parameters.address_n,
+                address_parameters.script_payment_hash,
+                address_parameters.address_type,
+                payment_credential_policy,
+            )
+            await show_set_output_stake_credential(
+                ctx,
+                address_parameters.address_n_staking,
+                address_parameters.staking_key_hash,
+                address_parameters.script_staking_hash,
+                address_parameters.certificate_pointer,
+                address_parameters.address_type,
+                stake_credential_policy,
+            )
+        else:
+            if _should_hide_output(address_parameters.address_n):
```

Comment:
Previously we showed warnings about the output being "unusual" even if it was going to be hidden. That is why the `_should_hide_output` function is called only if the address policy is simple, because we otherwise want to warn the user about the output being unusual.

I'll try to change it up a little. I think we don't need the `_should_hide_output` function at all.

Reply:
I've added a `_should_show_output` function which purely determines if the output should be shown. `_show_output` is called only if the output should be shown and is IMHO cleaner now. https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/ca9a89b2b753fbf0d744fb3bbec1335081ed6089#diff-eaac9c946526e07b3ac5c317971467dc330f41ee2cd7da91e701aa3098365bdbR701